exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/jpa/dao/LoginHistoryDAO.java

```diff
@@ -0,0 +1,324 @@
+package org.exoplatform.platform.gadget.services.LoginHistory.jpa.dao;
+
+import org.exoplatform.commons.api.persistence.ExoTransactional;
+import org.exoplatform.commons.persistence.impl.GenericDAOJPAImpl;
+import org.exoplatform.container.ExoContainerContext;
+import org.exoplatform.platform.gadget.services.LoginHistory.LastLoginBean;
+import org.exoplatform.platform.gadget.services.LoginHistory.LoginCounterBean;
+import org.exoplatform.platform.gadget.services.LoginHistory.LoginHistoryBean;
+import org.exoplatform.platform.gadget.services.LoginHistory.jpa.entity.LoginHistoryEntity;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.organization.OrganizationService;
+
+import java.sql.Timestamp;
+import java.util.*;
+
+public class LoginHistoryDAO extends GenericDAOJPAImpl {
```

Comment:
```suggestion
public class LoginHistoryDAO extends GenericDAOJPAImpl<LoginHistoryEntity, Long> {
```
The type of entities managed by this DAO must be added in the class declarative in order to type the operations.

Reply:
Okey i will fix it